src/common/xattr.h

```diff
@@ -19,12 +19,14 @@
 extern "C" {
 #endif
 
+#if defined(__linux__)
```

Comment:
i am inclined to keep it as it is, unless there is case that we see redefinition issues or warnings. why we could have redefinition issues because of this `#define`?

i don't think the whole piece should go. see 886f5a91f35d04cd4a721f1a66a2ef4a2c39fd5e